File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    numsLength := len(nums)
    numMap := make(map[int]int)
    for i := 0; i < numsLength; i++ {
        complement := target - nums[i]
        if j, ok := numMap[complement]; ok && j != i {
            return []int{i, j}
        }
        numMap[nums[i]] = i
    }
    return nil
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        /*
        for (int i = 0; i < nums.size() - 1; i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    return {i, j};
                }
            }
        }

        return {};
        */

        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];

            if (map.find(complement) != map.end()) {
                return {i, map[complement]};
            }

            map[nums[i]] = i; 
        }

        return {};
    }
};